insights/parsers/xfs_quota.py

```diff
@@ -0,0 +1,93 @@
+"""
+xfs_quota Commands
+==================
+
+Parser contains in this module is:
+
+XFSQuotaStatue - command ``/sbin/xfs_quota -x -c 'state -gu'``
+--------------------------------------------------------------
+"""
+
+from insights.core import Parser
+from insights.core.exceptions import SkipComponent
+from insights.core.plugins import parser
+from insights.parsers import get_active_lines
+from insights.specs import Specs
+
+
+@parser(Specs.xfs_quota_statue)
+class XFSQuotaStatue(Parser):
+    """
+    The ``/sbin/xfs_quota -x -c 'state -gu'`` command provides information for the
+    xfs quota devices.
+
+    Attributes:
+        group_quota (list): List of information for each group quota
+        user_quota (list): List of information for each user quota
+
+    Sample directory list collected::
+
+        User quota state on /sdd (/dev/sdd)
+          Accounting: ON
+          Enforcement: ON
+          Inode: #131 (1 blocks, 1 extents)
+        Blocks grace time: [7 days]
+        Blocks max warnings: 5
+        Inodes grace time: [7 days]
+        Inodes max warnings: 5
+        Realtime Blocks grace time: [7 days]
+        Group quota state on /sdd (/dev/sdd)
+          Accounting: OFF
+          Enforcement: OFF
+          Inode: N/A
+        Blocks grace time: [--------]
+        Blocks max warnings: 0
+        Inodes grace time: [--------]
+        Inodes max warnings: 0
+        Realtime Blocks grace time: [--------]
+        User quota state on /sdb (/dev/sdb)
+          Accounting: ON
+          Enforcement: ON
+          Inode: #131 (2 blocks, 2 extents)
+        Blocks grace time: [6 days]
+        Blocks max warnings: 10
+        Inodes grace time: [6 days]
+        Inodes max warnings: 5
+        Realtime Blocks grace time: [6 days]
+        Group quota state on /sdb (/dev/sdb)
+          Accounting: ON
+          Enforcement: ON
+          Inode: #137 (2 blocks, 2 extents)
+        Blocks grace time: [7 days]
+        Blocks max warnings: 5
+        Inodes grace time: [7 days]
+        Inodes max warnings: 5
+        Realtime Blocks grace time: [7 days]
+
+    Examples:
+        >>> type(quota_state)
+        <class 'insights.parsers.xfs_quota.XFSQuotaStatue'>
+        >>> len(quota_state.user_quota)
+        3
+        >>> quota_state.user_quota[0] == {'device': '/dev/sdd', 'accounting': 'ON', 'enforcement': 'ON', 'blocks_grace_time': '7 days', 'blocks_max_warnings': '5', 'inodes_grace_time': '7 days', 'inodes_max_warnings': '5', 'realtime_blocks_grace_time': '7 days'}
+        True
+    """
+
+    def parse_content(self, content):
+        self.user_quota = []
+        self.group_quota = []
+        data = None
+        for line in get_active_lines(content):
```

Comment:
As currently, performance is considered in high priority, it's not recommended to use the `get_active_lines` against `content`, especially if it's got from a command (it's assumed that there are no comment lines, e.g. `# xxx`. Better skip the empty lines in the following loop by hand, e.g. `if not line.strip(); continue`